src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -50,6 +50,314 @@ static void Main()
+    {
+        new C(a: 1, 2);
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "1 2.", parseOptions: TestOptions.Regular7_2);
+            verifier.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestSimpleConstructorWithOldLangVersion()
+        {
+            var source = @"
+class C
+{
+    C(int a, int b)
+    {
+        System.Console.Write($""{a} {b}."");
+    }
+    static void Main()
+    {
+        new C(a: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_1);
+            comp.VerifyDiagnostics(
+                // (10,21): error CS1738: Named argument specifications must appear after all fixed arguments have been specified. Please use language version 7.2 or greater to allow non-trailing named arguments.
+                //         new C(a: 1, 2);
+                Diagnostic(ErrorCode.ERR_NamedArgumentSpecificationBeforeFixedArgument, "2").WithArguments("7.2").WithLocation(10, 21)
+                );
+        }
+
+        [Fact]
+        public void TestSimpleExtension()
+        {
+            var source = @"
+public static class Extension
+{
+    public static void M(this C c, int a, int b)
+    {
+        System.Console.Write($""{a} {b}."");
+    }
+}
+public class C
+{
+    static void Main()
+    {
+        var c = new C();
+        c.M(a: 1, 2);
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "1 2.", parseOptions: TestOptions.Regular7_2, additionalRefs: new[] { SystemCoreRef });
+            verifier.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestSimpleExtensionWithOldLangVersion()
+        {
+            var source = @"
+public static class Extension
+{
+    public static void M(this C c, int a, int b)
+    {
+        System.Console.Write($""{a} {b}."");
+    }
+}
+public class C
+{
+    static void Main()
+    {
+        var c = new C();
+        c.M(a: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_1, references: new[] { SystemCoreRef });
+            comp.VerifyDiagnostics(
+                // (14,19): error CS1738: Named argument specifications must appear after all fixed arguments have been specified. Please use language version 7.2 or greater to allow non-trailing named arguments.
+                //         c.M(a: 1, 2);
+                Diagnostic(ErrorCode.ERR_NamedArgumentSpecificationBeforeFixedArgument, "2").WithArguments("7.2").WithLocation(14, 19)
+                );
+        }
+
+        [Fact]
+        public void TestSimpleDelegate()
+        {
+            var source = @"
+class C
+{
+    delegate void MyDelegate(int a, int b);
+    event MyDelegate e;
+
+    static void M(int a, int b)
+    {
+        System.Console.Write($""{a} {b}."");
+    }
+
+    static void Main()
+    {
+        var c = new C();
+        c.e += M;
+        c.e.Invoke(a: 1, 2);
```

Comment:
Is this binding through a different code path that regular method calls? If not, consider testing `c.e(a: 1, 2)` as well.